File: src/scripts/order_data.py
# coding: utf-8
# (C) Júlio Barreto
# Script para ordenar os dados do data.js por nome
# Criado em 16/02/2019
# Atualizado em 03/03/2019

import json, os

cwd = os.getcwd()

def update_file(name):
    with open(name, 'r') as datajs:
        """
        Abre o arquivo atual de dados e extrai o json de acronimos
        """
        datafile = json.load(datajs)

    with open(name, 'w') as newdatajs:
        """
        Exporta o arquivo de dados de modo que as siglas estejam ordenadas
        """
        json_text = json.dumps(datafile, sort_keys=True, indent=4, ensure_ascii=False).encode('utf8')
        newdatajs.write(json_text + '\n')

path_to_data = '/src/lib/%s.json'
data_files = ['cursos', 'disciplinas', 'girias', 'locais', 'outros']

print('\033[1;93m Reordenando arquivos de dados \033[0m')

for df in data_files:
    update_file((cwd + path_to_data) % df)

File: src/scripts/orderData.js
// Script original em python por Júlio Barreto,
// traduzido para javascript por Paulo Leitão.

const fs = require('fs');

const dataFiles = ['cursos', 'disciplinas', 'girias', 'locais', 'outros'];

function sortFile(file) {
  let orderedFile = {};
  Object.keys(file)
    .sort()
    .forEach(key => {
      orderedFile[key] = file[key];
    });

  return orderedFile;
}

function updateFile(name) {
  const path = `src/lib/${name}.json`;
  const rawDataFile = fs.readFileSync(path);
  const dataFile = JSON.parse(rawDataFile);
  const orderedDataFile = sortFile(dataFile);
  const dataString = JSON.stringify(orderedDataFile, null, 4) + '\n';
  fs.writeFileSync(path, dataString);
}

dataFiles.forEach(df => updateFile(df));
